### ISSUE TITLE
App crashes when showing a Dialog

### ISSUE THREAD

**Issue by theapache64** [JetBrains/compose-jb#444]
**⬇️ Input**

```kotlin
@Composable
fun DialogButton() {
    var shouldShowDialog by remember { mutableStateOf(false) }
    Button(
        onClick = {
            shouldShowDialog = true
        }
    ) {
        Text(text = "SHOW DIALOG")
    }

    if (shouldShowDialog) {

        Dialog(
            onDismissRequest = {
                shouldShowDialog = false
            }
        ) {
            Text(text = "Sample content")
        }
    }
}
```

😊 **Expected Behaviour**

- Dialog appears when the button clicked

😐 **Actual Behaviour**

- App crashing when `SHOW DIALOG` button pressed

💥 **Crash log**

<details>
<summary>Click to view</summary>

```
12:54:43 PM: Executing task 'run'...

> Task :compileKotlin
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :inspectClassesForKotlinIC
> Task :jar

> Task :run
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007fa61dd405e9, pid=18367, tid=18399
#
# JRE version: OpenJDK Runtime Environment (15.0.2+7) (build 15.0.2+7-27)
# Java VM: OpenJDK 64-Bit Server VM (15.0.2+7-27, mixed mode, sharing, tiered, compressed oops, g1 gc, linux-amd64)
# Problematic frame:
# C  [libskiko-linux-x64.so+0x2d25e9]  SkPictureRecorder::finishRecordingAsPicture()+0x29
#
# Core dump will be written. Default location: Core dumps may be processed with "/usr/share/apport/apport %p %s %c %d %P %E" (or dumping to /home/theapache64/Documents/projects/cd-distributable-test/core.18367)
#
# An error report file with more information is saved as:
# /home/theapache64/Documents/projects/cd-distributable-test/hs_err_pid18367.log
#
# If you would like to submit a bug report, please visit:
#   https://bugreport.java.com/bugreport/crash.jsp
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#

> Task :run FAILED
4 actionable tasks: 4 executed

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':run'.
> Process 'command '/home/theapache64/.jdks/openjdk-15.0.2/bin/java'' finished with non-zero exit value 134

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 9s
12:54:52 PM: Task execution finished 'run'.

```
</details>

💻 **Full Source Code**

- https://github.com/theapache64/cd-distributable-test/tree/dialog-issue

**EDIT**

OS: **Ubuntu 16.04**

**Comment by dector** [comment]
I have the same issue on `OpenJDK Runtime Environment (11.0.9.1+1) (build 11.0.9.1+1-post-Debian-1deb10u2)`

**Comment by jimgoog** [comment]
Presumably you are running into this on Linux?  I tested on Mac and it seems to show the dialog without issue.

**Comment by theapache64** [comment]
Yes. Linux. Ubuntu 16.04

**Comment by dzirbel** [comment]
I have the same issue on Ubuntu 20.10, with `OpenJDK Runtime Environment (11.0.10+9) (build 11.0.10+9-Ubuntu-0ubuntu1.20.10)`. Occurs on both `0.3.2` and `0.4.0-build171`.

It only seems to happen when the dialog state _changes_ - it works if showing the dialog on the first composition, but crashes if it's either shown in a later recomposition or if dismissed.

<details>
<summary>Crash log in case it's helpful</summary>

```
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007f9b3e8445e9, pid=42912, tid=42965
#
# JRE version: OpenJDK Runtime Environment (11.0.10+9) (build 11.0.10+9-Ubuntu-0ubuntu1.20.10)
# Java VM: OpenJDK 64-Bit Server VM (11.0.10+9-Ubuntu-0ubuntu1.20.10, mixed mode, sharing, tiered, compressed oops, g1 gc, linux-amd64)
# Problematic frame:
# C  [libskiko-linux-x64.so+0x2d25e9]  SkPictureRecorder::finishRecordingAsPicture()+0x29
#
# Core dump will be written. Default location: Core dumps may be processed with "/usr/share/apport/apport %p %s %c %d %P %E" (or dumping to /home/dzirbel/workspace/spotify/core.42912)
#
# An error report file with more information is saved as:
# /home/dzirbel/workspace/spotify/hs_err_pid42912.log
Compiled method (c1)   95729 4232   !   3       org.jetbrains.skija.PictureRecorder::finishRecordingAsPicture (31 bytes)
 total in heap  [0x00007f9c1d8bd790,0x00007f9c1d8be0a8] = 2328
 relocation     [0x00007f9c1d8bd908,0x00007f9c1d8bd988] = 128
 main code      [0x00007f9c1d8bd9a0,0x00007f9c1d8bde40] = 1184
 stub code      [0x00007f9c1d8bde40,0x00007f9c1d8bde90] = 80
 oops           [0x00007f9c1d8bde90,0x00007f9c1d8bde98] = 8
 metadata       [0x00007f9c1d8bde98,0x00007f9c1d8bdec8] = 48
 scopes data    [0x00007f9c1d8bdec8,0x00007f9c1d8bdf90] = 200
 scopes pcs     [0x00007f9c1d8bdf90,0x00007f9c1d8be050] = 192
 dependencies   [0x00007f9c1d8be050,0x00007f9c1d8be060] = 16
 handler table  [0x00007f9c1d8be060,0x00007f9c1d8be0a8] = 72
Compiled method (c1)   95730 3930       3       org.jetbrains.skija.impl.Managed$CleanerThunk::run (31 bytes)
 total in heap  [0x00007f9c1d7cf590,0x00007f9c1d7d0468] = 3800
 relocation     [0x00007f9c1d7cf708,0x00007f9c1d7cf7f0] = 232
 main code      [0x00007f9c1d7cf800,0x00007f9c1d7d00e0] = 2272
 stub code      [0x00007f9c1d7d00e0,0x00007f9c1d7d0160] = 128
 oops           [0x00007f9c1d7d0160,0x00007f9c1d7d0178] = 24
 metadata       [0x00007f9c1d7d0178,0x00007f9c1d7d01e0] = 104
 scopes data    [0x00007f9c1d7d01e0,0x00007f9c1d7d0300] = 288
 scopes pcs     [0x00007f9c1d7d0300,0x00007f9c1d7d0440] = 320
 dependencies   [0x00007f9c1d7d0440,0x00007f9c1d7d0458] = 24
 nul chk table  [0x00007f9c1d7d0458,0x00007f9c1d7d0468] = 16
Compiled method (c1)   95730 4795   !   3       androidx.compose.runtime.BroadcastFrameClock::sendFrame (116 bytes)
 total in heap  [0x00007f9c1da2b710,0x00007f9c1da2c4a0] = 3472
 relocation     [0x00007f9c1da2b888,0x00007f9c1da2b918] = 144
 main code      [0x00007f9c1da2b920,0x00007f9c1da2c140] = 2080
 stub code      [0x00007f9c1da2c140,0x00007f9c1da2c1b0] = 112
 oops           [0x00007f9c1da2c1b0,0x00007f9c1da2c1b8] = 8
 metadata       [0x00007f9c1da2c1b8,0x00007f9c1da2c1c0] = 8
 scopes data    [0x00007f9c1da2c1c0,0x00007f9c1da2c2f0] = 304
 scopes pcs     [0x00007f9c1da2c2f0,0x00007f9c1da2c3d0] = 224
 dependencies   [0x00007f9c1da2c3d0,0x00007f9c1da2c3d8] = 8
 handler table  [0x00007f9c1da2c3d8,0x00007f9c1da2c480] = 168
 nul chk table  [0x00007f9c1da2c480,0x00007f9c1da2c4a0] = 32
Compiled method (c1)   95731 3481       3       androidx.compose.ui.platform.DesktopOwners::invalidateIfNeeded (135 bytes)
 total in heap  [0x00007f9c1d6ac690,0x00007f9c1d6adaf8] = 5224
 relocation     [0x00007f9c1d6ac808,0x00007f9c1d6ac8f8] = 240
 main code      [0x00007f9c1d6ac900,0x00007f9c1d6ad720] = 3616
 stub code      [0x00007f9c1d6ad720,0x00007f9c1d6ad7b0] = 144
 oops           [0x00007f9c1d6ad7b0,0x00007f9c1d6ad7b8] = 8
 metadata       [0x00007f9c1d6ad7b8,0x00007f9c1d6ad7e8] = 48
 scopes data    [0x00007f9c1d6ad7e8,0x00007f9c1d6ad908] = 288
 scopes pcs     [0x00007f9c1d6ad908,0x00007f9c1d6ada98] = 400
 dependencies   [0x00007f9c1d6ada98,0x00007f9c1d6adaa8] = 16
 nul chk table  [0x00007f9c1d6adaa8,0x00007f9c1d6adaf8] = 80
#
# If you would like to submit a bug report, please visit:
#   https://bugs.launchpad.net/ubuntu/+source/openjdk-lts
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#
```
</details>

**Comment by theapache64** [comment]
@olonho @igordmn Any workaround for this issue? :grimacing:

**Comment by igordmn** [comment]
> @olonho @igordmn Any workaround for this issue? 😬

Just one guess. Will disabling GC fix the issue?

Add "–XX:+DisableExplicitGC" to Run configuration:

![image](https://user-images.githubusercontent.com/5963351/111004471-28f0bd00-839a-11eb-8e05-a5d13eb42808.png)

![image](https://user-images.githubusercontent.com/5963351/111004493-35751580-839a-11eb-9960-b7235ae7f620.png)

**Comment by dzirbel** [comment]
> Just one guess. Will disabling GC fix the issue?

No, this didn't work for me.

**Comment by theapache64** [comment]
Me neither.

**Comment by jershell** [comment]
I have the same issue 
OS: Ubuntu 20.04, 
JAVA: openjdk 14.0.2 2020-07-14
OpenJDK Runtime Environment AdoptOpenJDK (build 14.0.2+12)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 14.0.2+12, mixed mode, **sharing)
Compose version: 0.3.2 ; 0.4.0-build174

**Comment by igordmn** [comment]
Should be fixed in 0.4.0-build175

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-5479">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
